add: guides for writing and verifying contracts (#56)
maticnetwork/zkevm-docs: docs/guides/verify-smart-contract.md

---
id: verify-smart-contract
title: Verifying Smart Contracts
sidebar_label: Verify Contracts
description: A guide to verifying smart contracts in various frameworks on Polygon zkEVM network.
keywords:
  - docs
  - polygon
  - developer
  - guides
  - verify smart contract
  - zkEVM
---

> Once a smart contract is deployed to zkEVM, it can be verified in various ways depending on the framework of deployment as well the complexity of the contract. The aim here to use examples to illustrate how you can manually verify a deployed smart contract. 

Author: ```suggestion
Once a smart contract is deployed to zkEVM, it can be verified in various ways depending on the framework of deployment as well as the complexity of the contract. The aim here is to use examples to illustrate how you can manually verify a deployed smart contract. 
```

    - Save this file to parse it with [Prettier](https://prettier.io/)

    - Find the input JSON object. It will look [something like this](https://docs.soliditylang.org/en/latest/using-the-compiler.html#input-description) &rarr; `"input": {}`

    - Copy the **input** object value into a new file

    - **Name and save** this file locally in the **root folder**. Check this [example file](https://github.com/oceans404/zkevm-hardhat-demo/blob/main/example-standard-input.json) for reference

    - Drag and drop the **Standard Input JSON** file into *Drop the standard input JSON file or Click here* field. Once pasted, the **Verify & Publish** button becomes active.

3. Since you have provided an input, set *Try to fetch constructor arguments automatically* to **No**.

4. To add your ABI-encoded constructor arguments: 

    - Open the [Online ABI Encoder](https://abi.hashex.org/)  
    - Choose the `Auto-parse` tab.
    - Copy the ABI-encoded output.
    - Paste it into `ABI-encoded Constructor Arguments` if required by the contract.

</details>

> 9. Once you psate the contents of the newly created `.sol` file to the *Enter the Solodity Contract* field, the **Verify & Publish** button will be active.

Author: ```suggestion
8. Once you paste the contents of the newly created `.sol` file to the *Enter the Solidity Contract* field, the **Verify & Publish** button will be active.
```

> 10. Click on **Verify & Publish** to verify your deployed smart contract.

Author: ```suggestion
9. Click on **Verify & Publish** to verify your deployed smart contract.
```
